faculty_cli/cli.py

```diff
@@ -1224,6 +1224,12 @@ def sync_down(project, remote, local, server, rsync_opts):
 @click.argument("path")
 def ls(project, path):
     """List files and directories on Faculty workspace."""
+    try:
+        root_child = path.split("/")[1]
+    except IndexError:
+        _print_and_exit("{}: Not a valid path".format(path), 64)
+    if root_child is None or root_child != "project":
+        _print_and_exit("{} is outside the project workspace".format(path), 66)
 
```

Comment:
This looks like a reasonable change, but might something like the following not be more readable:

```python
if not path.startswith("/project/"):
    _print_and_exit("{} is not in the project workspace".format(path), 66)
```

Reply:
Agreed. I've changed this in the code.